### PR TITLE
[type:fix]fix namespace delete

### PR DESCRIPTION
**Pull request by xcsnx**
Make sure that:

- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/NamespaceServiceImpl.java [hunk 1]
```diff
@@ -55,12 +66,32 @@ public class NamespaceServiceImpl implements NamespaceService {
 
     private PluginService pluginService;
 
+    private SelectorMapper selectorMapper;
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/NamespaceServiceImpl.java:69]: pls use private final xxxService/Mapper xxservice/xxmapper

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/NamespaceServiceImpl.java [hunk 2]
```diff
@@ -83,10 +114,35 @@ public String delete(final List<String> ids) {
         if (ids.contains(Constants.DEFAULT_NAMESPACE_PRIMARY_KEY)) {
             return AdminConstants.SYS_DEFAULT_NAMESPACE_ID_DELETE;
         }
-        List<NamespaceDO> namespaceDOS = namespaceMapper.selectByIds(ids);
-        if (CollectionUtils.isEmpty(namespaceDOS)) {
+        Optional<NamespaceDO> namespaceDOS = namespaceMapper.selectByIds(ids).stream().findFirst();
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/NamespaceServiceImpl.java:117]: this is batch operation, not findFirst

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/NamespaceServiceImpl.java (hunk 2 above)
Comment (xcsnx) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/NamespaceServiceImpl.java:117]: ok

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/NamespaceServiceImpl.java [hunk 3]
```diff
@@ -83,10 +113,34 @@ public String delete(final List<String> ids) {
...
             return AdminConstants.SYS_DEFAULT_NAMESPACE_ID_DELETE;
         }
-        List<NamespaceDO> namespaceDOS = namespaceMapper.selectByIds(ids);
-        if (CollectionUtils.isEmpty(namespaceDOS)) {
+        List<String> namespaceIdList = namespaceMapper.selectByIds(ids).stream().map(NamespaceDO::getNamespaceId).collect(Collectors.toList());
+        if (namespaceIdList.isEmpty()) {
             return AdminConstants.SYS_NAMESPACE_ID_NOT_EXIST;
         }
+        List<NamespacePluginVO> namespacePluginVOS = namespacePluginRelMapper.selectAllByNamespaceIds(namespaceIdList);
+        if (!namespacePluginVOS.isEmpty()) {
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/NamespaceServiceImpl.java:121]: CollectionUtils.isNotEmpty

Comment (moremind) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/NamespaceServiceImpl.java:121]: change other conditions

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/NamespaceServiceImpl.java (hunk 3 above)
Comment (xcsnx) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/NamespaceServiceImpl.java:121]: ok

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/NamespaceServiceImpl.java (hunk 1 above)
Comment (moremind) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/NamespaceServiceImpl.java:69]: pls change this, add `final` for these fields

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/NamespaceServiceImpl.java [hunk 4]
```diff
@@ -83,10 +114,34 @@ public String delete(final List<String> ids) {
...
+        if (CollectionUtils.isNotEmpty(metaDataDOList)) {
+            return "metaData exist under those namespace!";
+        }
+        List<AuthPathDO> authPathDOList = authPathMapper.findByNamespaceIds(namespaceIdList);
+        if (CollectionUtils.isNotEmpty(authPathDOList)) {
+            return "authPath exist under those namespace!";
+        }
+        List<DiscoveryDO> discoveryDOList = discoveryMapper.selectAllByNamespaceIds(namespaceIdList);
+        if (CollectionUtils.isNotEmpty(discoveryDOList)) {
+            return "discovery exist under those namespace!";
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/NamespaceServiceImpl.java:143]: may throw exception

**Review by moremind — CHANGES_REQUESTED** [review]

**Review by moremind — APPROVED** [review]

**Pull request merged**